### PR TITLE
Move components input to separate class

### PR DESCRIPTION
**Pull request by roman-simionov**

### CODE REVIEW

File: js/renovation/widget.tsx [hunk 1]
```diff
@@ -55,7 +57,7 @@ const getAttributes = ({ elementAttr, accessKey }) => {
     return attrs;
 };
 
-const getCssClasses = (model: Partial<Widget>) => {
+const getCssClasses = (model: Partial<Widget> & Partial<WidgetInput>) => {
```
Comment (MaximKudriavtsev) [inline js/renovation/widget.tsx:60]: Can we use just `model: Partial<Widget>` here?

Comment (roman-simionov) [reply js/renovation/widget.tsx:60]: No, we pass an object that contains a mix of props and internal states
https://github.com/DevExpress/DevExtreme/blob/2177399ec14e7d4a4b455bfcfcb196535353c8cf/js/renovation/widget.tsx#L107

Comment (MaximKudriavtsev) [reply js/renovation/widget.tsx:60]: As I understand, we combine internal state and input props into the Widget class.

Comment (roman-simionov) [reply js/renovation/widget.tsx:60]: We need using `this.props.member` to access to a member of ComponentInput
https://github.com/DevExpress/DevExtreme/blob/2177399ec14e7d4a4b455bfcfcb196535353c8cf/js/renovation/widget.tsx#L82

**Review by churkin — DISMISSED** [review]

**Review by LazyLahtak — DISMISSED** [review]

**Review by LazyLahtak — APPROVED** [review]

**Review by MaximKudriavtsev — APPROVED** [review]

File: js/renovation/button.tsx [hunk 2]
```diff
@@ -19,10 +16,10 @@ const getImageContainerJSX = (source: string) => {
 const stylingModes = ['outlined', 'text', 'contained'];
 const defaultClassNames = ['dx-button'];
 
-const getCssClasses = (model: any) => {
+const getCssClasses = (model: ButtonInput) => {
     const { text, icon, stylingMode, type } = model;
-    const classNames = defaultClassNames.concat(model.classNames);
-    const isValidStylingMode = stylingModes.indexOf(stylingMode) !== -1;
+    const classNames = defaultClassNames.concat(model.classNames || []);
+    const isValidStylingMode = stylingMode && stylingModes.indexOf(stylingMode) !== -1;
```
Comment (churkin) [inline js/renovation/button.tsx:22]: `stylingModes?.indexOf(stylingMode)`

**Review by churkin — APPROVED** [review]

File: js/renovation/button.tsx (hunk 2 above)
Comment (roman-simionov) [reply js/renovation/button.tsx:22]: We will get `Argument of type '"outlined" | "text" | "contained" | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.ts(2345)` in this case

**Review by pavelgruba — APPROVED** [review]

**Pull request merged**